SwiftAppTemplate/SwiftAppTemplate/MainViewController.swift

```diff
@@ -0,0 +1,171 @@
+            collectPaymentButton.centerXAnchor.constraint(equalTo: containerView.centerXAnchor),
+            collectPaymentButton.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            collectPaymentButton.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: MainViewController.spacing),
+            collectPaymentButton.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -MainViewController.spacing),
+        ])
+
+        loadingContainerView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingContainerView.topAnchor.constraint(equalTo: view.topAnchor),
+            loadingContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            loadingContainerView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            loadingContainerView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+        ])
+
+        loadingIndicator.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingIndicator.centerXAnchor.constraint(equalTo: loadingContainerView.centerXAnchor),
+            loadingIndicator.centerYAnchor.constraint(equalTo: loadingContainerView.centerYAnchor),
+        ])
+
+        loadingLabel.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingLabel.topAnchor.constraint(equalTo: loadingIndicator.bottomAnchor, constant: MainViewController.spacing),
+            loadingLabel.centerXAnchor.constraint(equalTo: loadingContainerView.centerXAnchor)
+        ])
+    }
+
+
+    @objc func discoverReader(sender: UIButton) {
+        loadingLabel.text = "Discovering Readers..."
+        loadingContainerView.isHidden = false
+        loadingIndicator.startAnimating()
+
+        // TODO once you have discovered some readers create a
+        // ReaderListViewController and push it onto the navigationController.
+
+        // let readerViewController = ReaderListViewController(readers)
+        // navigationController?.pushViewController(readerViewController, animated: true)
+    }
+
+    @objc func collectPayment(sender: UIButton) {
+        startCollectingPayment()
+    }
+
+    // Mark: - UITextFieldDelegate
+
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        startCollectingPayment()
+        return true
+    }
+
+    private func startCollectingPayment() {
+        guard let amountString = paymentTextField.text else {
+            errorAlert("Please input an amount to collect.")
+            return
+        }
+
+        guard let amountNumber = Int(amountString) else {
+            errorAlert("Please enter valid integer.")
+            return
+        }
+
+        // TODO: Fill in flow for collecting payments. Since there are multiple steps
```

Comment:
nit: again, maybe better to not give too much guidance here, and remove the comment?